tests/src/Kernel/Access/RolesAndPermissionsUiAccessTest.php

```diff
@@ -0,0 +1,324 @@
+              'entity_type_id' => 'block_content',
+              'bundle_id' => 'group',
+            ],
+          ],
+          // The permissions form for administrators of a single group type.
+          [
+            'og_ui.permissions_edit_form',
+            [
+              'entity_type_id' => 'block_content',
+              'bundle_id' => 'group',
+              'role_name' => OgRoleInterface::ADMINISTRATOR,
+            ],
+          ],
+          // The permissions form for non-members of a single group type.
+          [
+            'og_ui.permissions_edit_form',
+            [
+              'entity_type_id' => 'block_content',
+              'bundle_id' => 'group',
+              'role_name' => OgRoleInterface::ANONYMOUS,
+            ],
+          ],
+          // The permissions form for members of a single group type.
+          [
+            'og_ui.permissions_edit_form',
+            [
+              'entity_type_id' => 'block_content',
+              'bundle_id' => 'group',
+              'role_name' => OgRoleInterface::AUTHENTICATED,
+            ],
+          ],
+          // The permissions form for "moderators" (a custom role) of a single
+          // group type.
+          [
+            'og_ui.permissions_edit_form',
+            [
+              'entity_type_id' => 'block_content',
+              'bundle_id' => 'group',
+              'role_name' => 'moderator',
+            ],
+          ],
+          // The overview of available roles for a group type.
+          [
+            'entity.og_role.collection',
+            [
+              'entity_type_id' => 'block_content',
+              'bundle_id' => 'group',
+            ],
+          ],
+          // The form to add a new role to a group type.
+          [
+            'entity.og_role.add_form',
+            [
+              'entity_type_id' => 'block_content',
+              'bundle_id' => 'group',
+            ],
+          ],
+          // The form to edit a custom role belonging to a group type.
+          [
+            'entity.og_role.edit_form',
+            [
+              'og_role' => 'block_content-group-moderator',
+            ],
+          ],
+          // The form to delete a custom role belonging to a group type.
+          [
+            'entity.og_role.delete_form',
+            [
+              'og_role' => 'block_content-group-moderator',
+            ],
+          ],
+        ],
+        [
+          // Since these routes are for managing the roles and permissions of
+          // all groups of the tested entity type and bundle, the forms should
+          // only be accessible to the root user, global administrators that
+          // have all permissions, and people with the 'administer organic groups'
+          // permission. Group administrators should not have access to these
+          // pages, but they will have access to the forms that deal with
+          // group-specific roles and permissions. These are not tested here.
```

Comment:
```suggestion
          // Since these routes are for managing the roles and permissions of
          // all groups of the tested entity type and bundle, the forms should
          // only be accessible to the root user, global administrators that
          // have all permissions, and users that have the permission
          // 'administer organic groups'.
          // Group administrators should not have access to these
          // pages, but they will have access to the forms that deal with
          // group-specific roles and permissions. These are not tested here.
```